common/src/main/java/com/alibaba/nacos/common/notify/NotifyCenter.java

```diff
@@ -0,0 +1,289 @@
+        if (consumer instanceof SmartSubscriber) {
+            EventPublisher.SMART_SUBSCRIBERS.remove((SmartSubscriber) consumer);
+            return;
+        }
+        if (SlowEvent.class.isAssignableFrom(cls)) {
+            INSTANCE.sharePublisher.unSubscriber(consumer);
+            return;
+        }
+        final String topic = consumer.subscriberType().getCanonicalName();
+        if (INSTANCE.publisherMap.containsKey(topic)) {
+            EventPublisher publisher = INSTANCE.publisherMap.get(topic);
+            publisher.unSubscriber(consumer);
+            return;
+        }
+        throw new NoSuchElementException("The subcriber has no event publisher");
+    }
+
+    /**
+     * request publisher publish event
+     * Publishers load lazily, calling publisher. Start () only when the event is actually published
+     *
+     * @param event
+     */
+    public static boolean publishEvent(final Event event) {
+        try {
+            return publishEvent(event.getClass(), event);
+        } catch (Throwable ex) {
+            LOGGER.error("There was an exception to the message publishing : {}", ex);
+            return false;
+        }
+    }
+
+    /**
+     * request publisher publish event
+     * Publishers load lazily, calling publisher. Start () only when the event is actually published
+     *
+     * @param eventType
+     * @param event
+     */
+    private static boolean publishEvent(final Class<? extends Event> eventType,
+                                        final Event event) {
+        final String topic = eventType.getCanonicalName();
+        if (SlowEvent.class.isAssignableFrom(eventType)) {
+            return INSTANCE.sharePublisher.publish(event);
+        }
+
+        if (INSTANCE.publisherMap.containsKey(topic)) {
+            EventPublisher publisher = INSTANCE.publisherMap.get(topic);
+            return publisher.publish(event);
+        }
+        throw new NoSuchElementException(
+            "There are no [" + topic + "] publishers for this event, please register");
+    }
+
+    /**
+     * register to share-publisher
+     *
+     * @param supplier
+     * @param eventType
+     * @return
+     */
+    public static EventPublisher registerToSharePublisher(
+        final Class<? extends SlowEvent> eventType) {
+        return INSTANCE.sharePublisher;
+    }
+
+    /**
+     * register publisher
+     *
+     * @param supplier
+     * @param eventType
+     * @param queueMaxSize
+     * @return
+     */
+    public static EventPublisher registerToPublisher(final Class<? extends Event> eventType,
+                                                     final int queueMaxSize) {
+        if (SlowEvent.class.isAssignableFrom(eventType)) {
+            return INSTANCE.sharePublisher;
+        }
+
+        final String topic = eventType.getCanonicalName();
+        INSTANCE.publisherMap.putIfAbsent(topic, BUILD_FACTORY.apply(eventType, queueMaxSize));
```

Comment:
computeIfAbsent  is better，lazy create value

Reply:
if jdk6 not support，I think you can implement in MapUtils

Reply:
Okay, this issue will be resolve in next commit.It may be later.